src/Modules/Guilds/Hooks/useTimeDetail.ts

```diff
@@ -0,0 +1,51 @@
+import moment from 'moment';
+import { useTranslation } from 'react-i18next';
+import { ProposalState } from 'types/types.guilds.d';
+import {
+  getTimeDifferenceFromCurrentTimeHumanized,
+  isBeforeCurrentTime,
+} from 'utils/time/time';
+import { useGuildConfig } from './useGuildConfig';
+
+const useTimeDetail = (
+  guildId: string,
+  status: ProposalState,
+  endTime: moment.Moment
+) => {
+  const { t } = useTranslation();
+  const { data: guildConfig } = useGuildConfig(guildId);
+
+  let endTimeDetail = null;
+  let endTimeMoment = null;
+
+  if (!endTime) return { endTimeMoment, endTimeDetail };
+
+  // with states Executable and Failed we show the time difference with execution time. Otherwise, with ending time
+  if (

```

Comment:
I think `guildConfig` always returns an object at least with empty properties and this will thus the first part of the if-statement will always be true. FYI

You could do `guildConfig?.timeForExecution`to be more specific